### PR TITLE
jemalloc: disable transparent hugepages by default on ARMv6/7

### PR DESCRIPTION
**Pull request by lopsided98**
###### Motivation for this change

This PR disables transparent hugepage support in jemalloc by default on ARMv6/7 (adding a parameter that allows it to be enabled if desired). The NixOS ARMv6/7 kernels do not enable transparent hugepage support, but jemalloc is not able to detect this (see jemalloc/jemalloc#526).

Even though the `--disable-thp` flag is available in jemalloc 4.5.0, it does not work correctly in this version, so this PR also updates jemalloc to 5.0.1 (which requires a large rebuild with the possibility of breakage).

Alternatively, this problem could be solved by enabling THP in the default kernel config, but it still might be good to have an option to disable it in jemalloc.

Also, is `stdenv.system` the correct way to do platform detection nowadays (with the upcoming better cross compiling support)?

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dezgeg** [comment]
I think enabling THP on 32-bit ARM is not possible by default, given that it depends on LPAE:

````
1692 config HAVE_ARCH_TRANSPARENT_HUGEPAGE
1693        def_bool y
1694        depends on ARM_LPAE
````

File: pkgs/development/libraries/jemalloc/default.nix [hunk 1]
```diff
@@ -1,18 +1,24 @@
-{ stdenv, fetchurl }:
+{ stdenv, fetchurl,
+  # jemalloc is unable to correctly detect transparent hugepage support on
+  # ARM, and the default kernel ARMv6/7 kernel does not enable it, so we
+  # explicitly disable support
+  thpSupport ? stdenv.system != "armv7l-linux" &&
```
Comment (dezgeg) [inline pkgs/development/libraries/jemalloc/default.nix]: Use: `stdenv.isArm`.

Comment (flokli) [reply pkgs/development/libraries/jemalloc/default.nix]: I'd also add a link to https://github.com/jemalloc/jemalloc/issues/526 in the comments, so we know when we can strip this.

File: pkgs/development/libraries/jemalloc/default.nix [hunk 2]
```diff
@@ -1,18 +1,23 @@
-{ stdenv, fetchurl }:
+{ stdenv, fetchurl,
+  # jemalloc is unable to correctly detect transparent hugepage support on
+  # ARM (https://github.com/jemalloc/jemalloc/issues/526), and the default
+  # kernel ARMv6/7 kernel does not enable it, so we explicitly disable support
+  thpSupport ? !stdenv.isArm }:
```
Comment (Mic92) [inline pkgs/development/libraries/jemalloc/default.nix]: I would not expose this as an option.

**Pull request merged**